e2e/tests/contracts.rs

```diff
@@ -1993,27 +1999,77 @@ async fn simulations_can_be_made_without_coins() -> Result<()> {
     .contract_id;
 
     let provider = wallet.provider().clone();
-    let no_funds_wallet = Wallet::random(&mut thread_rng(), provider.clone());
+    let no_funds_wallet = Wallet::random(&mut thread_rng(), provider);
 
     let response = MyContract::new(contract_id, no_funds_wallet)
         .methods()
         .get(5, 6)
-        .simulate(Execution::StateReadOnly)
+        .simulate(Execution::state_read_only())
         .await?;
 
     assert_eq!(response.value, 11);
 
     Ok(())
 }
 
+#[tokio::test]
+async fn simulations_can_be_made_at_specific_block_height() -> Result<()> {
+    abigen!(Contract(
+        name = "MyContract",
+        abi = "e2e/sway/contracts/contract_test/out/release/contract_test-abi.json"
+    ));
+
+    let wallets = setup_node_with_high_price(true).await?;
+    let wallet = wallets.first().expect("has wallet");
+
+    let contract_id = Contract::load_from(
+        "sway/contracts/contract_test/out/release/contract_test.bin",
+        LoadConfiguration::default(),
+    )?
+    .deploy_if_not_exists(wallet, TxPolicies::default())
+    .await?
+    .contract_id;
```

Comment:
You can use setup program test if you wish, just make the wallets before the macro.